content/applications/general/voip/axivox/dial_plan_advanced.rst

```diff
@@ -0,0 +1,289 @@
+navigating to :guilabel:`Switches` in the left menu. To create a new switch click :guilabel:`Add a
+switch` and then configure a :guilabel:`Name` for it.
+
+Toggle the :guilabel:`Switch` :guilabel:`On` and :guilabel:`Off` from the :guilabel:`State` column
+in the Switches dashboard (click :guilabel:`Switches` in the left menu).
+
+This :guilabel:`On` / :guilabel:`Off` state automatically routes traffic in a dial plan in which
+this switch is set. The traffic travels to the :guilabel:`Active` route when :guilabel:`On` is
+toggled in the switch. The call traffic travels to the :guilabel:`Inactive` route when
+:guilabel:`Off` is toggled in the switch. Changes can be made on the fly, just be sure to click
+:guilabel:`Apply changes` to implement the them.
+
+Add a switch to dial plan
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+To add a :guilabel:`Switch` to a dial plan, navigate to `Axivox management console
+<https://manage.axivox.com>`_ and click on :guilabel:`Dial plans` in the left menu. Then select or
+create a dial plan, and click on the drop-down menu. Select :guilabel:`Switch` and then click
+:guilabel:`Add`. Double click on the element to set the :guilabel:`Switch`.
+
+.. image:: dial_plan_advanced/switch.png
+   :align: center
+   :alt: Switch configuration in a dial plan, with inactive and active routes highlighted.
+
+Multi-switch
+------------
+
+A :guilabel:`Multi-switch` in Axivox is a switch where multiple paths can be configured and switched
+between. To configure and set a :guilabel:`Multi-switch`, navigate to `Axivox management console
+<https://manage.axivox.com>`_. Then, click on the :guilabel:`Switches` menu item in the left menu.
+Toggle to the :guilabel:`Multi-switch` tab to create or set a pre-configured
+:guilabel:`Multi-switch`.
+
+To create a new :guilabel:`Multi-switch` click :guilabel:`Create new`. Enter a :guilabel:`Name` for
+the :guilabel:`Multi-switch` and then enter the :guilabel:`Available choice`. Enter one
+:guilabel:`Available choice` per line. Do **not** duplicate any entries.
+
+To select the :guilabel:`State` of the :guilabel:`Multi-switch` click the drop-down menu next to the
+:guilabel:`Multi-switch` name. Whichever :guilabel:`State` is chosen is the route that is followed
+in the dial plan. The :guilabel:`State` can be edited on the fly, just be sure to :guilabel:`Apply
+changes`.
+
+Add a multi-switch to dial plan
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+To add a :guilabel:`Multi-switch` to a dial plan, first, navigate to `Axivox management console
+<https://manage.axivox.com>`_ and click on :guilabel:`Dial plans` in the left menu. Then select or
+create a dial plan, and click on the drop-down menu. Select :guilabel:`Multi-switch` and then click
+:guilabel:`Add`. Double click on the element to set the :guilabel:`Switch`.
+
+.. image:: dial_plan_advanced/multi-switch.png
+   :align: center
+   :alt: Multi-switch configuration in a dial plan, with chosen route highlighted.
```

Comment:
```suggestion
Access list scenario
--------------------

An :guilabel:`Access List` element in a dial plan allows for the routing of certain numbers, and
disallows (denies) other numbers.

After adding an :guilabel:`Access List` element to a dial plan, it can be configured by
double-clicking on the element directly in the :guilabel:`Dialplan Editor` pop-up window.

Two fields appear where regular expressions can based in the :guilabel:`Allow` and :guilabel:`Deny`
fields of the :guilabel:`Access List` pop-up window.

.. example::
   For a very important customer, their number can be set in the :guilabel:`Allow` field, and these
   callers can be sent directly to management.

.. tip::
   A regular expression (shortened to regex or regexp), sometimes also referred to as a rational
   expression, is a sequence of characters that specifies a match pattern in text.

.. image:: dial_plan_advanced/access-config.png
   :align: center
   :alt: Access list element configuration with the allow/deny fields highlighted.

.. example::
   - `2\\d\\d`: validates numbers from 200 to 299
   - `02\\d*`: validates all numbers beginning with 02
   - `0017165551212`: validates the number (0017165551212)

After setting the :guilabel:`Allow` and :guilabel:`Deny` fields with regular expressions or numbers,
click :guilabel:`Save` on the :guilabel:`Access List` pop-up window.

Then, on the :guilabel:`Access list` element in the dial plan, three paths (or routes) are available
to link to further actions.

Unknown calls can be routed through the regular menu flow by adding a :guilabel:`Menu` element, and
connecting it to the :guilabel:`Unknown` path. :guilabel:`Refused` calls can be routed to the
:guilabel:`Hang up` element. Lastly, :guilabel:`Authorized` callers can be sent to a specific
extension or queue.

.. image:: dial_plan_advanced/access-list.png
   :align: center
   :alt: Access list element highlighted in an example dial plan.

Switches
========

A :guilabel:`Switch` element in Axivox is a simple activated/deactivated route action.

These can be activated or chosen quickly, allowing for quick routing changes, without altering the
dial plan.

Alternate routes can be set to jump to, at a moment's notice, based on variables, such as
availability, or to adjust traffic flow for any number of reasons.

Axivox allows for a simple on/off switch, and a multi-switch, which can have several paths to choose
from.

- :guilabel:`Switch`: a manual on/off control that can divert traffic, based on whether it is opened
  (On) or closed (Off).
- :guilabel:`Multi-Switch`: a mechanism to create paths, and turn them on and off, to divert
  incoming calls.

Basic switch
------------

A :guilabel:`Switch` can be set in the `Axivox management console <https://manage.axivox.com>`_ by
navigating to :guilabel:`Switches` in the left menu. To create a new switch click :guilabel:`Add a
switch`, configure a :guilabel:`Name` for it, and click :guilabel:`Save`.

Then, toggle the desired switch to either :guilabel:`On` or :guilabel:`Off`, from the
:guilabel:`State` column in the :guilabel:`Switches`` dashboard (accessible via the
:guilabel:`Switches` option in the left menu).

This :guilabel:`On` / :guilabel:`Off` state automatically routes traffic in a dial plan, in which
this switch is set.

The traffic travels to the :guilabel:`Active` route when :guilabel:`On` is toggled in the switch.
The call traffic travels to the :guilabel:`Inactive` route when :guilabel:`Off` is toggled in the
switch.

Changes can be made on the fly, just be sure to click :guilabel:`Apply changes` to implement the
them.

Add a switch to dial plan
~~~~~~~~~~~~~~~~~~~~~~~~~

To add a :guilabel:`Switch` to a dial plan, navigate to `Axivox management console
<https://manage.axivox.com>`_, and click on :guilabel:`Dial plans` in the left menu. Then, click
:guilabel:`Visual Editor` next to the desired dial plan to open the :guilabel:`Dialplan Editor`
pop-up window. 

Then, from the :guilabel:`New element` drop-down menu, select :guilabel:`Switch`, and then click
:guilabel:`Add`. Double-click on the element to further configure the :guilabel:`Switch` element.

.. image:: dial_plan_advanced/switch.png
   :align: center
   :alt: Switch configuration in a dial plan, with inactive and active routes highlighted.

Multi-switch
------------

A :guilabel:`Multi-Switch` element in Axivox is a switch where multiple paths can be configured, and
switched between.

To configure, and set, a :guilabel:`Multi-Switch` element, navigate to `Axivox management console
<https://manage.axivox.com>`_. Then, click on the :guilabel:`Switches` menu item in the left menu.

Toggle to the :guilabel:`Multi-switch` tab to create, or set, a pre-configured
:guilabel:`Multi-Switch` element.

To create a new :guilabel:`Multi-Switch`, click :guilabel:`Create new`. Then, enter a
:guilabel:`Name` for the element, and then enter the :guilabel:`Available choice`. Enter one
:guilabel:`Available choice` per line. Do **not** duplicate any entries.

Remember to click :guilabel:`Save` when done.

To select the :guilabel:`State` of the :guilabel:`Multi-switch`, click the drop-down menu next to
the :guilabel:`Multi-switch` name, under the :guilabel:`Multi-switch` tab on the
:guilabel:`Switches` page.

The :guilabel:`State` chosen is the route that is followed in the dial plan. The :guilabel:`State`
can be edited on the fly, just be sure to click :guilabel:`Apply changes`.

Add a multi-switch to dial plan
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

To add a :guilabel:`Multi-Switch` element to a dial plan, navigate to `Axivox management console
<https://manage.axivox.com>`_, and click :guilabel:`Dial plans` in the left menu.

Then, select or create a dial plan. Next, click :guilabel:`Visual Editor` on the desired dial plan.

On the :guilabel:`Dialplan Editor` pop-up window that appears, click on the :guilabel:`New element`
drop-down menu, and select :guilabel:`Multi-Switch`. Then, click :guilabel:`Add`.

Double-click on the element to further configure the :guilabel:`Switch` element.

.. image:: dial_plan_advanced/multi-switch.png
   :align: center
   :alt: Multi-switch configuration in a dial plan, with chosen route highlighted.
```